script/Base.s.sol

```diff
@@ -58,14 +64,48 @@ contract BaseScript is Script, Sphinx {
         }
         sphinxProjectName = vm.envOr({ name: "SPHINX_PROJECT_NAME", defaultValue: TEST_SPHINX_PROJECT_NAME });
 
-        // Avalanche has a lower block gas limit than most other chains.
-        if (block.chainid == AVALANCHE_CHAIN_ID) {
-            maxSegmentCount = 300;
-            maxTrancheCount = 298;
-        } else {
-            maxSegmentCount = 500;
-            maxTrancheCount = 500;
-        }
+        // Avalanche chain ID.
+        segmentsCountMap[43_114] = 560;
+        tranchesCountMap[43_114] = 570;
+
+        // Base chain ID.
+        segmentsCountMap[8453] = 2220;
+        tranchesCountMap[8453] = 2310;
+
+        // Blast chain ID.
+        segmentsCountMap[238] = 1120;
+        tranchesCountMap[238] = 1160;
+
+        // BSC chain ID.
+        segmentsCountMap[56] = 4890;
+        tranchesCountMap[56] = 5200;
+
+        // Ethereum chain ID.
+        segmentsCountMap[1] = 1120;
+        tranchesCountMap[1] = 1160;
+
+        // Gnosis chain ID.
+        segmentsCountMap[100] = 630;
+        tranchesCountMap[100] = 650;
+
+        // Optimism chain ID.
+        segmentsCountMap[10] = 1120;
+        tranchesCountMap[10] = 1160;
+
+        // Polygon chain ID.
+        segmentsCountMap[137] = 1120;
+        tranchesCountMap[137] = 1160;
+
+        // Scroll chain ID.
+        segmentsCountMap[534_352] = 370;
+        tranchesCountMap[534_352] = 380;
+
+        // Sepolia chain ID.
+        segmentsCountMap[11_155_111] = 1120;
+        tranchesCountMap[11_155_111] = 1160;
```

Comment:
Which one should we go with?

1. Keep this map here so the the value of count is automatically picked during the deployments.
2. Remove this map from here, maintain a table in `benchmarks` folder and add count manually during deployments.